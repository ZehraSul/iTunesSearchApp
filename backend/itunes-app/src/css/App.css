/* Styling for app.js */

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.App {
  margin-top: 5%;
  text-align: center;
  margin-left: 1.5%;
}

.App h1 {
  margin-bottom: 3%;
  font-size: 6vw;
  background: linear-gradient(to right, #ff5d51 5%, #9671fc 50%, #03c7fd 95%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.App hr {
  width: 75%;
  margin: 50px auto 0px auto;
}
