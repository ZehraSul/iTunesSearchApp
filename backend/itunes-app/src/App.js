import { useEffect, useState } from "react";
import "./css/App.css";
import ListFavourites from "./Components/ListFavourites";
import ListResults from "./Components/ListResults";
import Search from "./Components/Search";
import { FAVOURITES_URL } from "./config/config";

function App() {
  //Setting intial state
  const [newSearch, setNewSearch] = useState([]);
  const [searchString, newSearchString] = useState("");
  const [searchType, newSearchType] = useState("");
  const [favourites, setFavourites] = useState([]);

  // Making an API call on button click
  const SearchHandler = () => {
    fetch(
      `${FAVOURITES_URL}search?term=${searchString}&media=${searchType}&limit=10`,
      {
        headers: {
          Accept: "application/json",
          "Content-Type": "application/json",
        },
      }
    )
      .then((res) => {
        return res.json();
      })
      .then(
        (data) => {
          // Replacing stardard image size with larger ones
          data.results.forEach((r) => {
            r.artworkUrl600 = r.artworkUrl100.replace("100x100", "600x600");
          });
          setNewSearch(data.results);
        },
        (err) => {
          console.log(err);
        }
      );
  };

  // Calls API to get items that have been favourited
  useEffect(() => {
    fetch(FAVOURITES_URL, {
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
      },
    })
      .then((res) => {
        return res.json();
      })
      .then(
        (data) => {
          setFavourites(data);
        },
        (err) => {
          console.log(err);
        }
      );
  }, []);

  //Displaying various components and passing though data child components
  return (
    <div className="App">
      <h1>iTunes Search App</h1>
      <Search
        searchString={searchString}
        newSearchString={newSearchString}
        searchType={searchType}
        newSearchType={newSearchType}
        SearchHandler={SearchHandler}
      />
      <ListResults
        newSearch={newSearch}
        setNewSearch={setNewSearch}
        favourites={favourites}
        setFavourites={setFavourites}
      />
      <hr />
      <ListFavourites favourites={favourites} setFavourites={setFavourites} />
    </div>
  );
}

export default App;
