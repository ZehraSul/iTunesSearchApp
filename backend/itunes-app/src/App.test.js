import renderer from "react-test-renderer";
import App from "./App";

// Snapshot test
test("renders correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
