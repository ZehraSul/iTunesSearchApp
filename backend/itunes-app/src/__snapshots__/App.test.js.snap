// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <h1>
    iTunes Search App
  </h1>
  <div>
    <div
      className="mb-3 input-group"
      style={
        Object {
          "margin": "auto",
          "paddingTop": "2%",
          "width": "50%",
        }
      }
    >
      <input
        aria-label="Text input with dropdown button"
        className="form-control"
        onChange={[Function]}
        placeholder="Search"
      />
      <select
        align="end"
        className="form-select"
        id="input-group-dropdown-2"
        onChange={[Function]}
        variant="outline-secondary"
      >
        <option
          disabled={true}
          hidden={true}
          selected={true}
          value=""
        >
          Media Type
        </option>
        <option
          value="movie"
        >
          Movies
        </option>
        <option
          value="podcast"
        >
          Podcasts
        </option>
        <option
          value="music"
        >
          Music
        </option>
        <option
          value="musicVideo"
        >
          Music Video
        </option>
        <option
          value="shortFilm"
        >
          Short Film
        </option>
        <option
          value="tvShow"
        >
          TV Show
        </option>
        <option
          value="ebook"
        >
          eBook
        </option>
        <option
          value="all"
        >
          All
        </option>
      </select>
      <button
        className="btn btn-danger"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <h2>
      Results
    </h2>
    <p
      className="text-muted"
    >
      Nothing to show
    </p>
  </div>
  <hr />
  <div
    className="row"
  >
    <h2>
      Favourites
    </h2>
    <p
      className="text-muted"
    >
      Nothing to show
    </p>
  </div>
</div>
`;
